### PR TITLE
replace FileSystems with ZipInputStream for parallel execution

### PR DESCRIPTION
**Pull request by atty303**
The steps to reproduce the problem are as follows.

1. Parallel execution is enabled.
2. Targets that use `scalaPBUnpackProto` will be executed in parallel.
3. `FileSystemAlreadyExistsException` will be thrown.

This is because FileSystems does not allow multiple instances of the same URI to be created.
Therefore, I have changed the implementation to use ZipInputStream.

### CODE REVIEW

File: contrib/scalapblib/src/ScalaPBModule.scala [hunk 1]
```diff
@@ -94,35 +93,27 @@ trait ScalaPBModule extends ScalaModule {
...
-                }
-                super.visitFile(f, a)
+    val buffer = new Array[Byte](8192)
+    cp.iterator.foreach { ref =>
+      Using(new ZipInputStream(ref.path.getInputStream)) { zip =>
+        var entry = Option(zip.getNextEntry)
+        while (entry.nonEmpty) {
+          val name = entry.get.getName
+          if (name.endsWith(".proto")) {
+            val protoDest = os.Path(os.RelPath(name), dest)
```
Comment (lefou) [inline contrib/scalapblib/src/ScalaPBModule.scala:103]: I'd suggest `os.Subpath` which ensures we won't step outside the parent dir.

```suggestion
            val protoDest = dest / os.SubPath(name)
```

File: contrib/scalapblib/src/ScalaPBModule.scala [hunk 2]
```diff
@@ -94,35 +93,27 @@ trait ScalaPBModule extends ScalaModule {
...
+            Using(os.write.outputStream(protoDest, createFolders = true)) { o =>
+              var r = 0
+              while (r != -1) {
+                r = zip.read(buffer)
+                if (r != -1) o.write(buffer, 0, r)
               }
```
Comment (lefou) [inline contrib/scalapblib/src/ScalaPBModule.scala:109]: You may want to use `mill.api.IO.stream` here.

**Review by lefou — COMMENTED** [review]
Looks good to me. See my suggestion to use more mill API. Is this covered by any test?

**Comment by atty303** [comment]
Thanks for the review!
I have fixed the points you pointed out.

> Looks good to me. See my suggestion to use more mill API. Is this covered by any test?

I don't think there are any tests related to this. I have tested it working on a project at work.

**Review by lefou — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**